src/connectedk8s/azext_connectedk8s/_precheckutils.py

```diff
@@ -0,0 +1,219 @@
+            try:
+                # Checking if job get scheduled or not
+                if event["object"].metadata.name == "pre-onboarding-inspector-job":
+                    is_job_scheduled = True
+                # Checking if job reached completed stage or not
+                if event["object"].metadata.name == "pre-onboarding-inspector-job" and event["object"].status.conditions[0].type == "Complete":
+                    is_job_complete = True
+                    w.stop()
+            except Exception as e:
+                continue
+            else:
+                continue
+
+        if (is_job_scheduled is False):
+            logger.warning("Unable to schedule the pre onboarding inspector job in the kubernetes cluster. The possible reasons can be presence of a security policy or security context constraint (SCC) or it may happen becuase of lack of ResourceQuota.\n")
+            Popen(cmd_helm_delete, stdout=PIPE, stderr=PIPE)
+            return
+        elif (is_job_scheduled is True and is_job_complete is False):
+            logger.warning("Unable to finish the pre onboarding inspector job in the kubernetes cluster. The possible reasons can be presence of lack of Resources on the cluster.\n")
+            Popen(cmd_helm_delete, stdout=PIPE, stderr=PIPE)
+            return
+        else:
+            # Fetching the pre onboarding inspector Container logs
+            all_pods = corev1_api_instance.list_namespaced_pod('default')
+            # Traversing through all agents
+            for each_pod in all_pods.items:
+                # Fetching the current Pod name and creating a folder with that name inside the timestamp folder
+                pod_name = each_pod.metadata.name
+                if(pod_name.startswith(job_name)):
+                    # Creating a text file with the name of the container and adding that containers logs in it
+                    preonboarding_inspector_container_log = corev1_api_instance.read_namespaced_pod_log(name=pod_name, container="pre-onboarding-inspector-container", namespace='default')
+        # Clearing all the resources after fetching the pre onboarding inspector container logs
+        Popen(cmd_helm_delete, stdout=PIPE, stderr=PIPE)
+
+    # To handle any exception that may occur during the execution
+    except Exception as e:
+        logger.warning("An exception has occured while trying to execute the pre onboarding inspector in the cluster. Exception: {}".format(str(e)) + "\n")
+        Popen(cmd_helm_delete, stdout=PIPE, stderr=PIPE)
+        telemetry.set_exception(exception=e, fault_type=consts.Pre_Onboarding_Inspector_Failed_Fault_Type, summary="Error while executing Pre onboarding inspector Job")
+        return
+
+    return preonboarding_inspector_container_log
+
+
+def helm_install_release(chart_path, http_proxy, https_proxy, no_proxy, proxy_cert, kube_config, kube_context, helm_client_location, onboarding_timeout="60"):
```

Comment:
let's rename it to helm_install_release_preonboarding_inspector ? since other files have the same name for another method